### PR TITLE
Bugfix/geinfra 116

### PR DESCRIPTION
**Pull request by DavidOberholzer**

### CODE REVIEW

File: swagger_server/controllers/access_control_controller.py [hunk 1]
```diff
@@ -879,11 +880,33 @@ def resource_list(offset=None, limit=None, prefix=None, resource_ids=None):  # n
...
+            query = query.filter(models.Resource.id.in_(resource_ids))
+
+    # Additional prefix filter startswith.
+    if prefix:
+        query = query.filter(models.Resource.urn.startswith(prefix))
+
+    query = query.order_by(models.Resource.id)
+
+    return db_actions.transform(
+        query.offset(
```
Comment (cobusc) [inline swagger_server/controllers/access_control_controller.py]: Can you mover the offset and limit out of the `transform()` call, please?

File: swagger_server/controllers/access_control_controller.py [hunk 2]
```diff
@@ -879,11 +880,33 @@ def resource_list(offset=None, limit=None, prefix=None, resource_ids=None):  # n
...
+                        getattr(models.Resource, key).in_(
+                            _id if isinstance(_id, list) else [_id]
+                        )
+                    )
+        else:
+            query = query.filter(models.Resource.id.in_(resource_ids))
+
+    # Additional prefix filter startswith.
+    if prefix:
+        query = query.filter(models.Resource.urn.startswith(prefix))
```
Comment (cobusc) [inline swagger_server/controllers/access_control_controller.py]: Nice!

File: swagger_server/controllers/access_control_controller.py [hunk 3]
```diff
@@ -879,11 +880,33 @@ def resource_list(offset=None, limit=None, prefix=None, resource_ids=None):  # n
...
     """
-    return db_actions.crud(
-        model="Resource",
-        api_model=Resource,
-        action="list",
-        query={"offset": offset, "limit": limit, "ids": resource_ids, "order_by": ["id"]}
+    # Mimicking db_actions.list_entry() with additional filter of prefix.
+    query = db.session.query(models.Resource, func.count().over().label("x_total_count"))
+    if resource_ids:
+        if isinstance(resource_ids, dict):
```
[the remark this answers is not in the document]
Comment (DavidOberholzer) [reply swagger_server/controllers/access_control_controller.py]: Ah yes, the dict is for handling composite keys I forgot.

File: swagger_server/controllers/access_control_controller.py (hunk 1 above)
Comment (DavidOberholzer) [reply swagger_server/controllers/access_control_controller.py]: sure

**Comment by cobusc** [comment]
Some cosmetic changes suggested. :+1: when done.

**Pull request merged**